lib/Doctrine/ODM/PHPCR/Translation/TranslationStrategy/AttributeTranslationStrategy.php

```diff
@@ -176,7 +176,7 @@ public function getLocalesFor($document, NodeInterface $node, ClassMetadata $met
         $locales = array();
         foreach ($node->getProperties("*{$this->prefix}*") as $prop) {
             $matches = null;
-            if (preg_match('/' . $this->prefix . ':(..)-[^-]*/', $prop->getName(), $matches)) {
+            if (preg_match('/' . $this->prefix . ':([a-z]{2}_?[a-z]{0,2})-[^-]*/', $prop->getName(), $matches)) {
```

Comment:
there might be a BC break in this, as the previous expressoin was way more open. i would prefer to keep that the same, so probably `((..)(_(a-zA-Z){2})?)-[^-]*/` - please also add some tests about this. we should get the full locale in the match i think. 

btw, why do we forbid further `-` in the property name? we should probably just stop matching after the `locale-` thing, we don't care what comes. as we don't have a `$` at the end, it does not matter anyways. so `((..)(_(a-zA-Z){2})?)-`

also, i think we should allow upper case letters for the sub locale, often those are written as de_AT